src/nvim/api/window.c

```diff
@@ -472,6 +472,48 @@ void nvim_win_config(Window window, Integer width, Integer height,
   }
 }
 
+/// Return window configuration of a window opened via |nvim_open_win()|.
+///
+/// Return a dictionary containing the same options that can be given to
+/// |nvim_open_win()| plus the height and the width of the window.
+///
+/// @param window Window handle
+/// @param[out] err Error details, if any
+/// @return Window configuration
+Dictionary nvim_win_get_config(Window window, Error *err)
+  FUNC_API_SINCE(6)
+{
+  Dictionary rv = ARRAY_DICT_INIT;
+
+  win_T *wp = find_window_by_handle(window, err);
+  if (!wp) {
+    return rv;
+  }
+
+  PUT(rv, "height", INTEGER_OBJ(wp->w_height));
+  PUT(rv, "width", INTEGER_OBJ(wp->w_width));
+  PUT(rv, "row", FLOAT_OBJ(wp->w_float_config.row));
+  PUT(rv, "col", FLOAT_OBJ(wp->w_float_config.col));
+  PUT(rv, "external", BOOLEAN_OBJ(wp->w_float_config.external));
```

Comment:
If `"external"` is set `"row", `"col"` and `"relative"` should be excluded. Can put them last for an early return.